galaxy/worker/logutils.py

```diff
@@ -62,9 +61,22 @@ def emit(self, record: logging.LogRecord) -> None:
 
         # TODO(cutwater): Revisit connection alias usage.
         models.ImportTaskMessage.objects.using('logging').create(
-            task_id=record.task_id,
-            message_type=constants.ImportTaskMessageType.from_logging_level(
+            task=record.task,
+            message_type=const.ImportTaskMessageType.from_logging_level(
                 record.levelno).value,
             message_text=record.msg,
             **create_kwargs,
         )
+
+
+class CollectionImportHandler(logging.Handler):
+    def emit(self, record: logging.LogRecord) -> None:
+        from galaxy.main import models
```

Comment:
Is this local import absolutely necessary?

Reply:
Currently it's necessary, because handlers are initialized as a part of logging config, which happens in settings and at that time django is not initialized and models imports at the module level fails.
I have an idea how to refactor this and move logging initialization for import tasks out of settings, but probably won't happen as this PR. I'll add TODO comment to refactor it in future.

Reply:
That's a fine reason. Certainly have hit that exact same limitation myself, numerous times. 👌